### PR TITLE
Feature/454 documentation refactor

### LINKED ISSUE #454: Refactor existing documentation
Change existing documentation for clarity.

### PR DESCRIPTION
**Pull request by YiFeiZhang2**
# Description

Update documentation to give more information about services, testing, authorization, and validation middleware.

Fixes #454, #394

## Type of change
- [x] This change requires a documentation update

### CODE REVIEW

File: docs/standards.md [hunk 1]
```diff
@@ -197,15 +198,140 @@ Things to take note of:
...
+            "confirmed": true,
+        }),
+    },
+    invalid: createNAccounts(10, {
+        "accountType": Constants.HACKER
+    })
+};
+```
+
+In this example the `new` accounts are accounts that exist, but the hacker objects have not been created. The `stored.team` accounts are those linked to a hacker object that is in a team. The `stored.noTeam` accounts link to  a hacker that is not in a team. The `invalid` accounts are created accounts that are linked to a hacker object that does not fit with the Hacker schema. The invalid accounts are used to test fail cases. The value for each key is an array of account objects.
```
Comment (pierreTklein) [inline docs/standards.md]: > The invalid accounts are used to test fail cases

What sort of fail cases? They aren't used for all fail cases.

Comment (pierreTklein) [inline docs/standards.md]: Add on the following sentence to this one:

> The `invalid` accounts are created accounts that are linked to a hacker object that does not fit with the Hacker schema. 

"These accounts are only invalid because the hacker object is malformed."

Comment (pierreTklein) [inline docs/standards.md]: > The value for each key is an array of account objects. 

Rephrase to:

> \`createNAccounts\` returns an array of account objects.

File: docs/standards.md [hunk 2]
```diff
@@ -197,15 +198,140 @@ Things to take note of:
...
+    })
+};
+```
+
+In this example the `new` accounts are accounts that exist, but the hacker objects have not been created. The `stored.team` accounts are those linked to a hacker object that is in a team. The `stored.noTeam` accounts link to  a hacker that is not in a team. The `invalid` accounts are created accounts that are linked to a hacker object that does not fit with the Hacker schema. The invalid accounts are used to test fail cases. The value for each key is an array of account objects. 
+
+On the other end of the account-hacker link, [hacker.util.js](../tests/util/hacker.test.util.js) contains the hacker data in the form of hacker objects. These hacker objects have an `accountId` attribute which references an account's `_id`. The matching between the hacker object and the respective account object it needs to link to is also done by nomenclature. For example, a hacker on a team would be called `TeamHackerX` where X is a number. This hacker's account object would be within the array specified by `hackerAccounts.stored.team`. The specific account object is referenced by its index in the array. That index is the same as the value X in the name of the hacker object.
```
Comment (pierreTklein) [inline docs/standards.md]: Include code for this example, with a value for X. It's hard to follow with just a verbal description.

File: docs/standards.md [hunk 3]
```diff
@@ -197,15 +198,140 @@ Things to take note of:
...
 
 ### Validation files
```
Comment (pierreTklein) [inline docs/standards.md]: What is a `validator` file?

File: docs/standards.md [hunk 4]
```diff
@@ -197,15 +198,140 @@ Things to take note of:
...
```
Comment (pierreTklein) [inline docs/standards.md]: What libraries do we use to perform validation? Where can I find external documentation for that?

File: docs/standards.md [hunk 5]
```diff
@@ -197,15 +198,140 @@ Things to take note of:
...
+        Middleware.Validator.RouteParam.idValidator,
+        Middleware.Validator.Account.updateAccountValidator,
+
+        Middleware.parseBody.middleware,
+        ...
+    );
+```
+
+`Middleware.parseBody.middleware` evaluates the validator functions, so it is generally placed right after the validators. If there is an error during validation, it will send a 422 error status to be handled. If all the validators pass, then it will move the validated values into req.body.
+
```
Comment (pierreTklein) [inline docs/standards.md]: Add a section about adding to `validator.helper`

File: docs/standards.md [hunk 6]
```diff
@@ -197,15 +198,140 @@ Things to take note of:
...
+        // validators
+        Middleware.Validator.RouteParam.idValidator,
+        Middleware.Validator.Account.updateAccountValidator,
+
+        Middleware.parseBody.middleware,
+        ...
+    );
+```
+
+`Middleware.parseBody.middleware` evaluates the validator functions, so it is generally placed right after the validators. If there is an error during validation, it will send a 422 error status to be handled. If all the validators pass, then it will move the validated values into req.body.
```
Comment (pierreTklein) [inline docs/standards.md]: > it will move the validated values into req.body.

Note that it will overwrite anything that is in `req.body`. So, if `req.body` is the following:
```json
{
  "A": "foo",
  "B": true,
  "C": { 
    "bar": "baz" 
  }
}
```
and we validate for only `A`, and `B`, then the `req.body` will have the following form after validation:
```json
{
  "A": "foo",
  "B": true
}
```

File: docs/standards.md [hunk 7]
```diff
@@ -197,15 +198,140 @@ Things to take note of:
...
+  * A boolean detailing whether this value is optional. `false` means that the value is necessary.
+
+A validator example:
+```javascript
+"use strict";
+const VALIDATOR = require("./validator.helper");
+const Constants = require("../../constants/general.constant");
+
+module.exports = {
+    inviteAccountValidator: [
```
Comment (pierreTklein) [inline docs/standards.md]: Make this validator array the same one that is used in the route example below

**Comment by loreina** [comment]
@pierreTklein @YiFeiZhang2 is this good to merge?
